include/envoy/event/dispatcher.h

```diff
@@ -75,11 +76,13 @@ class Dispatcher {
    * @param socket supplies an open file descriptor and connection metadata to use for the
    *        connection. Takes ownership of the socket.
    * @param transport_socket supplies a transport socket to be used by the connection.
+   * @param stream_info info object for the server connection
    * @return Network::ConnectionPtr a server connection that is owned by the caller.
    */
   virtual Network::ConnectionPtr
   createServerConnection(Network::ConnectionSocketPtr&& socket,
-                         Network::TransportSocketPtr&& transport_socket) PURE;
+                         Network::TransportSocketPtr&& transport_socket,
+                         StreamInfo::StreamInfo& stream_info) PURE;
```

Comment:
Given some of the other issues you have opened about async/blocking/etc. logging is this going to end up needing to be a shared_ptr? Not sure if it's worth it to do now or not. nbd

Reply:
Can we postpone adding shared pointer until we need it? I am not sure if it's the right granularity for capturing async logging.

Reply:
Yup sure.